lib/volcanic/authenticator/v1/warden/auth_strategy.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'warden'
+require 'forwardable'
+require_relative '../token'
+
+module Volcanic::Authenticator
+  module V1::Warden
+    # this class handle the auth service authentication by using warden
+    class AuthStrategy < Warden::Strategies::Base
+      private
+
+      def validate_token
+        token = Volcanic::Authenticator::V1::Token.new(auth_token)
+        if token.remote_validate
+          success!(token)
+        else
+          invalid!
+        end
+      rescue Volcanic::Authenticator::V1::TokenError
+        invalid!
+      rescue Volcanic::Authenticator::V1::ConnectionError => e
+        fail!(e)
+      end
+
+      def auth_token
+        @auth_token ||= env['HTTP_AUTHORIZATION'].to_s.gsub('Bearer ', '')
```

Comment:
consider `request.has_header?('authorization')` and `request.get_header('authorization')` methods which may help you

Reply:
seems like `get_header(name)` is calling `env(name)`  and rack request instance doesnt contains any value of `authorization`, only `HTTP_AUTHORIZATION`.